File: Scripts/volatile_and_force_compiler_task/call-an-object-method/javascript/call-an-object-method.js
class CallAnObjectMethod {
	constructor() {
	}

	someMethod(a, i) {
		var value = a + i;
	}
}

var r = 1;
const obj = new CallAnObjectMethod();

function executeTask(i) {

	obj.someMethod(1, i);
	return i;
}

for (var i = 0; i < 2000000000; ++i) {
	r = executeTask(i);
}

File: scripts/volatile_and_force_compiler_task/call-an-object-method/java/CallAnObjectMethod.java
public class CallAnObjectMethod {

  private static volatile int r = 1; 
  public CallAnObjectMethod(){
  }

  public void someMethod(int value, int i){
   this.r = value + i;
  }

	public static int executeTask(int i) {

		CallAnObjectMethod obj = new CallAnObjectMethod();
		obj.someMethod(1, i);
		return i;
	}

public static void main(String[] args){

	for (int i = 0; i < 2000000000; ++i) {
		r = executeTask(i);
	}
}

}
